client/lib/index.js

```diff
@@ -27,6 +27,14 @@ async function isGzip(filePath) {
 	return buf[0] === 0x1f && buf[1] === 0x8b && buf[2] === 0x08;
 }
 
+async function isZip(filepath) {
+	const buf = Buffer.alloc(4);
+
+	await fs.read(await fs.open(filepath, 'r'), buf, 0, 4, 0)
+
+	return buf[0] === 0x50 && buf[1] === 0x4B && (buf[2] === 0x03 || buf[2] === 0x05 || buf[2] === 0x07) && (buf[3] === 0x04 || buf[3] === 0x06 || buf[3] === 0x08);
```

Comment:
This line is too long and it hurts readability. It would also be helpful to have a comment documenting the signature we're checking against (or linking to a specification or reference of some sort) rather than a bunch of magic numbers.